markets/perps-market/contracts/modules/PerpsAccountModule.sol

```diff
@@ -54,40 +56,64 @@ contract PerpsAccountModule is IPerpsAccountModule {
         PerpsMarketFactory.Data storage perpsMarketFactory = PerpsMarketFactory.load();
 
         GlobalPerpsMarket.Data storage globalPerpsMarket = GlobalPerpsMarket.load();
-        globalPerpsMarket.validateCollateralAmount(synthMarketId, amountDelta);
+        globalPerpsMarket.validateCollateralAmount(collateralId, amountDelta);
         globalPerpsMarket.checkLiquidation(accountId);
 
         PerpsAccount.Data storage account = PerpsAccount.create(accountId);
         uint128 perpsMarketId = perpsMarketFactory.perpsMarketId;
 
-        PerpsAccount.validateMaxCollaterals(accountId, synthMarketId);
+        PerpsAccount.validateMaxCollaterals(accountId, collateralId);
 
         AsyncOrder.checkPendingOrder(account.id);
 
         if (amountDelta > 0) {
-            _depositMargin(perpsMarketFactory, perpsMarketId, synthMarketId, amountDelta.toUint());
+            _depositMargin(perpsMarketFactory, perpsMarketId, collateralId, amountDelta.toUint());
         } else {
             uint256 amountAbs = MathUtil.abs(amountDelta);
             // removing collateral
             account.validateWithdrawableAmount(
-                synthMarketId,
+                collateralId,
                 amountAbs,
                 perpsMarketFactory.spotMarket
             );
-            _withdrawMargin(perpsMarketFactory, perpsMarketId, synthMarketId, amountAbs);
+            _withdrawMargin(perpsMarketFactory, perpsMarketId, collateralId, amountAbs);
         }
 
         // accounting
-        account.updateCollateralAmount(synthMarketId, amountDelta);
+        account.updateCollateralAmount(collateralId, amountDelta);
+
+        emit CollateralModified(accountId, collateralId, amountDelta, ERC2771Context._msgSender());
+    }
+
+    function debt(uint128 accountId) external view override returns (uint256 accountDebt) {
+        Account.exists(accountId);
+        PerpsAccount.Data storage account = PerpsAccount.load(accountId);
+
+        accountDebt = account.debt;
+    }
+
+    // 1. call depositMarketUsd and deposit amount directly to core system
+    // 2. look up account and reduce debt by amount
+    // 3. transfer synth to sender
```

Comment:
Sorry i dont understand here, what synth is sent to sender?

Reply:
oops stray comment; updated.  initially i thought this function would also return to the caller the amt of synth but thats not the case.